File: restaurantservice/supply-service/src/main/java/pl/restaurant/supplyservice/api/response/InvoiceView.java
package pl.restaurant.supplyservice.api.response;

import lombok.*;
import pl.restaurant.supplyservice.api.request.Address;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class InvoiceView {
    private String nr;
    private Long restaurantId;
    private RestaurantShortInfo restaurantInfo;
    private LocalDate date;
    private String sellerName;
    private String buyerName;
    private Address sellerAddress;
    private Address buyerAddress;
    private String sellerNip;
    private String buyerNip;
    private LocalDate completionDate;
    private BigDecimal price;
    private List<GoodView> goods;
}

File: restaurant-app/src/app/model/invoice/invoice-view.ts
import {Address} from "../address";
import {GoodView} from "./good-view";
import {RestaurantShortInfo} from "../restaurant/restaurant-short-info";

export interface InvoiceView {
  nr: string;
  restaurantId: any;
  restaurantInfo: RestaurantShortInfo | null;
  date: Date;
  sellerName: string;
  buyerName: string;
  sellerAddress: Address;
  buyerAddress: Address;
  sellerNip: string;
  buyerNip: string;
  completionDate: Date;
  price: number;
  goods: GoodView[];
}
